modules/standard/IO.chpl

```diff
@@ -2975,7 +2989,8 @@ private inline proc _read_one_internal(_channel_internal:qio_channel_ptr_t,
 private inline proc _write_one_internal(_channel_internal:qio_channel_ptr_t,
                                         param kind:iokind,
                                         x:?t,
-                                        loc:locale?):syserr where _isIoPrimitiveTypeOrNewline(t) {
+                                        loc:locale?,
+                                        byte_esc=false):syserr where _isIoPrimitiveTypeOrNewline(t) {
```

Comment:
I might be missing / misunderstanding some case, but it seems to me that `byte_esc` should not be an argument to this function and instead it should consider `byte_esc=true` whenever `t==_bytes`.